### PR TITLE
Refactored Lambda code examples

### PR DESCRIPTION
**Pull request by Doug-AWS**
*Issue #, if available:*

*Description of changes:*

- Re-ordered crates in Cargo.toml
- Refactored Lambda code examples to use common example pattern
- Added change-java-runtime and list-all-function-runtimes code examples

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws/sdk/examples/lambda/src/bin/change-java-runtime.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    let client = Client::from_conf(config);
+
+    // Get function's runtime
+    let resp = client.list_functions().send().await?;
+
+    for function in resp.functions.unwrap_or_default() {
+        if arn == function.function_arn.unwrap() {
+            let rt = function.runtime.unwrap();
+            if rt == Runtime::Java11 || rt == Runtime::Java8 {
+                // Change it to Java8a12 (Corretto)
```
Comment (rcoh) [inline aws/sdk/examples/lambda/src/bin/change-java-runtime.rs:69]: this changes the runtime of _every_ lambda function which definitely seems like it goes against "don't surprise users with what an example does"

**Review by rcoh — CHANGES_REQUESTED** [review]

File: aws/sdk/examples/lambda/src/bin/change-java-runtime.rs (hunk 1 above)
Comment (Doug-AWS) [reply aws/sdk/examples/lambda/src/bin/change-java-runtime.rs:69]: I'll rename it as change-all-java-runtimes.rs.

File: aws/sdk/examples/lambda/src/bin/invoke-function.rs [hunk 2]
```diff
@@ -137,7 +137,9 @@ async fn main() {
         //
         // For our example, we will simply print that the function doesn't
         // exist and return a non-zero exit code to indicate the failure.
-        Err(SdkError::ServiceError { err, .. }) if err.is_resource_not_found_exception() => {
+        Err(SdkError::ServiceError { err, .. })
```
Comment (rcoh) [inline aws/sdk/examples/lambda/src/bin/invoke-function.rs:140]: you're on an old SDK, you need to update

Comment (rcoh) [reply aws/sdk/examples/lambda/src/bin/invoke-function.rs:140]: you can just back this out entirely, the code that was there before is the pattern we want to guide people towards

File: aws/sdk/examples/lambda/src/bin/list-all-function-runtimes.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        println!("Region:                {:?}", &region);
+    }
+
+    // Get list of available regions.
+    let config = ec2::Config::builder().region(region).build();
+    let ec2_client = ec2::Client::from_conf(config);
+    let resp = ec2_client.describe_regions().send().await;
+
+    for region in resp.unwrap().regions.unwrap_or_default() {
+        show_lambdas(region.region_name.unwrap()).await;
```
Comment (rcoh) [inline aws/sdk/examples/lambda/src/bin/list-all-function-runtimes.rs:71]: this is actually a pattern we want to discourage—we don't want people to create a client over and over again, customers should be sharing a client between requests.

Comment (Doug-AWS) [reply aws/sdk/examples/lambda/src/bin/list-all-function-runtimes.rs:71]: The client only returns information about the lambdas in its region.

**Review by rcoh — APPROVED** [review]

**Pull request merged**